sweagent/agent/models.py

```diff
@@ -272,6 +272,80 @@ def query(self, history: list[dict[str, str]]) -> str:
         self.update_stats(input_tokens, output_tokens)
         return response.choices[0].message.content
 
+class GroqModel(BaseModel):
+    MODELS = {
+        "llama2-70b-4096": {
+            "max_context": 4096,
+            "cost_per_input_token": 0,
+            "cost_per_output_token": 0,
+        },
+        "mixtral-8x7b-32768": {
+            "max_context": 32768,
+            "cost_per_input_token": 0,
+            "cost_per_output_token": 0,
+        },
+        "gemma-7b-it": {
+            "max_context": 8192,
+            "cost_per_input_token": 0,
+            "cost_per_output_token": 0,
+        },
+    }
+
+    SHORTCUTS = {
+        "groq-llama70b": "llama2-70b-4096",
+        "groq-mixtral8x7b": "mixtral-8x7b-32768",
+        "groq-gemma7b": "gemma-7b-it",
+    }
+
+    def __init__(self, args: ModelArguments, commands: list[Command]):
+        super().__init__(args, commands)
+
+        # Set Groq key
+        cfg = config.Config(os.path.join(os.getcwd(), "keys.cfg"))
+        self.client = OpenAI(api_key=cfg["GROQ_API_KEY"], base_url="https://api.groq.com/openai/v1/")
+
+    def history_to_messages(
+        self, history: list[dict[str, str]], is_demonstration: bool = False
+    ) -> list[dict[str, str]]:
+        """
+        Create `messages` by filtering out all keys except for role/content per `history` turn
+        """
+        # Remove system messages if it is a demonstration
+        if is_demonstration:
+            history = [entry for entry in history if entry["role"] != "system"]
+            return '\n'.join([entry["content"] for entry in history])
+        # Return history components with just role, content fields
+        return [
+            {k: v for k, v in entry.items() if k in ["role", "content"]}
+            for entry in history
+        ]
+
+    @retry(
+        wait=wait_random_exponential(min=1, max=15),
+        reraise=True,
+        stop=stop_after_attempt(3),
+        retry=retry_if_not_exception_type((CostLimitExceededError, RuntimeError)),
+    )
+    def query(self, history: list[dict[str, str]]) -> str:
```

Comment:
How similar is this and `history_to_messages` to the `OpenaAIModel` class? Because if that's just exactly the same, let's just subclass from there to cut down on duplicity.

Reply:
It is the same since I'm using the OpenAI() client because Groq is design to be compatible with OpenAi.
 https://console.groq.com/docs/openai